astroquery/query.py

```diff
@@ -463,3 +496,101 @@ def login(self, *args, **kwargs):
+            requests.codes.unavailable,
+            requests.codes.service_unavailable,
+            requests.codes.gateway_timeout,
+            requests.codes.request_timeout,
+            requests.codes.timeout,
+            requests.codes.precondition_failed,
+            requests.codes.precondition,
+            requests.codes.payment_required,
+            requests.codes.payment
+        """
+
+    retry_errors = [requests.codes.unavailable,
+                    requests.codes.service_unavailable,
+                    requests.codes.gateway_timeout,
+                    requests.codes.request_timeout,
+                    requests.codes.timeout,
+                    requests.codes.precondition_failed,
+                    requests.codes.precondition,
+                    requests.codes.payment_required,
+                    requests.codes.payment]
+
+    def __init__(self, retry=True, start_delay=1, *args, **kwargs):
+        """
+        ::param retry: set to False if retries not required
+        ::param start_delay: start delay interval between retries (default=1s).
+                Note that for HTTP 503, this code follows the retry timeout
+                set by the server in Retry-After
+        """
+        self.retry = retry
+        self.start_delay = start_delay
+        super(RetrySession, self).__init__(*args, **kwargs)
+
+    def send(self, request, **kwargs):
+        """
+        Send a given PreparedRequest, wrapping the connection to service in
+        try/except that retries on
+        Connection reset by peer.
+        :param request: The prepared request to send._session
+        :param kwargs: Any keywords the adaptor for the request accepts.
+        :return: the response
+        :rtype: requests.Response
+        """
+        # merge kwargs with env
+        proxies = kwargs.get('proxies') or {}
+        settings = self.merge_environment_settings(
+            request.url, proxies, kwargs.get('stream'), kwargs.get('verify'),
+            kwargs.get('cert'))
+        kwargs.update(settings)
+
+        # requests does not provide a default timeout, hence we might need
+        # to add it
+        if 'timeout' not in kwargs or kwargs['timeout'] is None:
+            kwargs['timeout'] = 120  # default time (sec) to wait for response
+
+        if self.retry:
+            current_delay = max(self.start_delay, DEFAULT_RETRY_DELAY)
+            current_delay = min(current_delay, MAX_RETRY_DELAY)
+            num_retries = 0
+            log.debug(
+                "Sending request {0} to server.".format(request))
+            current_error = None
+            while num_retries < MAX_NUM_RETRIES:
+                try:
+                    response = super(RetrySession, self).send(request,
+                                                              **kwargs)
+                    response.raise_for_status()
+                    return response
+                except requests.HTTPError as e:
```

Comment:
we try to avoid single char varnames, so much easier to grep/etc with a bit more length :)

```suggestion
                except requests.HTTPError as err:
```